hq_superset/utils.py

```diff
@@ -37,3 +47,51 @@ def get_role_name_for_domain(domain):
     # Prefix in-case domain name matches with known role names such as admin
     # Same prefix pattern as schema only by coincidence, not a must.
     return f"{DOMAIN_PREFIX}{domain}"
+
+
+def get_column_dtypes(datasource_defn):
+    """
+    Maps UCR column data types to Pandas data types.
+
+    See corehq/apps/userreports/datatypes.py for possible data types.
+    """
+    # TODO: How are array indicators handled in CSV export?
+    pandas_dtypes = {
+        'date': 'datetime64[ns]',
+        'datetime': 'datetime64[ns]',
+        'string': 'string',
+        'integer': 'int64',
+        'decimal': 'float64',
+        'small_integer': 'int8',  # TODO: Is this true?
+    }
+    column_dtypes = {'doc_id': 'string'}
```

Comment:
I think there is UUID field in Postgres? Though, I am not sure if Sqlalchemy/pandas has support for that. 

The main thing to be concerned about though is if the column will be mapped to be a bounded length string or unbounded length string. See https://docs.sqlalchemy.org/en/14/core/type_basics.html#sqlalchemy.types.Unicode which can be bounded.

Reply:
There is a UUID field in Postgres, but [SQLAlchemy](https://docs.sqlalchemy.org/en/14/core/type_basics.html) doesn't have a UUID data type.

Good call on the bounded/unbounded strings! I will check what type string columns end up with in Postgres.